src/bentoml/serve.py

```diff
@@ -968,11 +750,44 @@ def serve_grpc_production(
             bento_identifier,
             f"http://{log_metrics_host}:{metrics_port}",
         )
-    arbiter = create_standalone_arbiter(
-        watchers=watchers, sockets=list(circus_socket_map.values())
-    )
 
-    with track_serve(svc, production=True, serve_kind="grpc"):
+    arbiter_kwargs: dict[str, t.Any] = {
+        "watchers": watchers,
+        "sockets": list(circus_socket_map.values()),
+    }
+
+    if development_mode:
+
+        plugins = []
+
+        if reload:
+            if sys.platform == "win32":
+                logger.warning(
+                    "Due to circus limitations, output from the reloader plugin will not be shown on Windows."
+                )
+            logger.debug(
+                "--reload is passed. BentoML will watch file changes based on 'bentofile.yaml' and '.bentoignore' respectively."
+            )
+
+            # NOTE: {} is faster than dict()
+            plugins = [
+                # reloader plugin
+                {
+                    "use": "bentoml._internal.utils.circus.watchfilesplugin.ServiceReloaderPlugin",
+                    "working_dir": working_dir,
+                    "bentoml_home": bentoml_home,
+                },
+            ]
+
+        arbiter_kwargs["plugins"] = plugins
+        arbiter_kwargs["debug"] = True if sys.platform != "win32" else False
+        arbiter_kwargs["loggerconfig"] = SERVER_LOGGING_CONFIG
+        arbiter_kwargs["loglevel"] = "WARNING"
```

Comment:
It seems like we are having this branch the same twice for both grpc and HTTP serve function, should we refactor out to a shared function?

Reply:
done